### PR TITLE
Allow `username` as parameter when using auth objects

### PR DESCRIPTION
**Pull request by VictorVelarde**
Related with #1022 

This PR adds `.username` as param in auth (the same as in CARTO.js). Current `.user` is kept for backwards compatibility.

I have also updated the current guides and examples in master to display the preferred option: **username**

### CODE REVIEW

File: src/setup/auth-service.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ function checkAuth (auth) {
     if (!util.isObject(auth)) {
         throw new CartoValidationError(`${cvt.INCORRECT_TYPE} 'auth' property must be an object.`);
     }
-    auth.username = auth.user; // API adapter
+    auth.username = util.isDefined(auth.username) ? auth.username : auth.user; // backwards compatibility
```
Comment (davidmanzanares) [inline src/setup/auth-service.js]: `util.isDefined(` could we deprecate it?

Comment (VictorVelarde) [reply src/setup/auth-service.js]: I have just added it to utils... as I was using `!util.isUndefined` ... but I can remove it if you don't like it very much! don't you like that sugar 😄 ?

Comment (davidmanzanares) [reply src/setup/auth-service.js]: Well... I don't like it much. I think it is more verbose than directly comparing it, and, if the language provides a reasonable to do it, I prefer that over an util

Comment (VictorVelarde) [reply src/setup/auth-service.js]: I see your point, you prefer not using a new util. I prefer `isDefined` to `!isUndefined`, cause IMO negatives are a bit harder to understand than positives. 

But it is not a big deal, so I'm gonna remove the new method. 
I'll also flip the if, cause I always try to express conditionals as positives.

Thx for your review!

**Pull request merged**